compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -800,6 +800,21 @@ LogicalResult setTileAndFuseLoweringConfig(IREE::GPU::TargetAttr target,
       threadTileSizes[loopIndex] = 1;
     }
 
+    // Special case: if all the parallel dimensions are dynamic (and, just to be
+    // defensive, there aren't workgroup tile size constraints requested), we'll
+    // let the later tiling logic mask off excess lanes in the last workgroup
+    // since that's cheaper than using insufficient parallelism.
+    if (!distInfo.partitionableLoops.empty() && !lossFactor &&
+        llvm::all_of(distInfo.partitionableLoops, [&](int64_t shapeDim) {
+          return ShapedType::isDynamic(distInfo.loopBounds[shapeDim]) &&
+                 workgroupTileSizeMultiples[shapeDim] == 1;
+        })) {
```

Comment:
Optional: We could make this inner-most non-unit dimensions. There shouldn't be unit dims in practice because we try to fold them away, but if there was it'd trip up this check.

Reply:
That's a big stack of complexity that I don't expect to see very often in practice?